### PR TITLE
Add script to generate projects.jsonl

### PR DESCRIPTION
**Pull request by polm**
## Description

This PR adds a script to generate `projects.jsonl`, which is useful for automation related to this repo. It's the same as #150 with no changes to individual READMEs.

### Types of change

enhancement 

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [ ] I ran the `update` scripts in the `.github` folder, and all the configs and docs are up-to-date.

### CODE REVIEW

File: .github/update_projects_json.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
```
Comment (adrianeboyd) [inline .github/update_projects_json.py:1]: Should this be renamed to `update_projects_jsonl.py`?

Comment (polm) [reply .github/update_projects_json.py:1]: I could do that if that's more clear, though I think it's fine as is.

File: .github/update_projects_json.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+from spacy.cli._util import PROJECT_FILE, load_project_config
+from wasabi import msg
+import json
+import typer
+
+
+def main(root: Path = typer.Argument(Path.cwd(), help="Root path to look in")):
+    """
+    Update the projects.json file for the repo.
```
Comment (rmitsch) [inline .github/update_projects_json.py:10]:
```suggestion
    Update the projects.jsonl file for the repo.
```

File: .github/update_projects_json.py (hunk 1 above)
Comment (rmitsch) [reply .github/update_projects_json.py:1]: I agree with the renaming suggestion.

File: README.md [hunk 3]
```diff
@@ -67,3 +67,4 @@ contains several scripts:
 | [`update_docs.py`](.github/update_docs.py)                   | Update all auto-generated docs in the given root. Calls into [`spacy project document`](https://spacy.io/api/cli#project-document) and only replaces the auto-generated sections, not any custom content before or after. |
 | [`update_category_docs.py`](.github/update_category_docs.py) | Update the auto-generated `README.md` in the category directories listing the available project templates.                                                                                                                |
 | [`update_configs.py`](.github/update_configs.py)             | Update and auto-fill all `config.cfg` files included in the repo, similar to [`spacy init fill-config`](https://spacy.io/api/cli#init-fill-config). Can be used to keep the configs up to date with changes in spaCy.     |
+| [`update_projects_json.py`](.github/update_projects_json.py) | Update `projects.jsonl` file in the given root. Should be used at the root level of the repo.                                                                                                                             |
```
Comment (rmitsch) [inline README.md:70]: If you decide to rename:
```suggestion
| [`update_projects_jsonl.py`](.github/update_projects_jsonl.py) | Update `projects.jsonl` file in the given root. Should be used at the root level of the repo.                                                                                                                             |
```

File: projects.jsonl [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
... command to prepare the coreference data:\n\n```\nspacy project run prep-conll-data\n```\n\nAfter that you can execute `spacy project run all`.\n\nIf you already have CoNLL 2012 data prepared and concatenated into one file per split, you can specify the paths to the training, dev, and test files directly in `project.yml`, see the `vars` section. After doing so you can run `spacy project run all`.\n\n## Using the Trained Pipeline\n\nAfter training the pipeline (or downloading a pretrained version), you can load and use it like this:\n\n```\nimport spacy\nnlp = spacy.load(\"training/coref\")\n\ndoc = nlp(\"John Smith called from New York, he says it's raining in the city.\")\n# check the word clusters\nprint(\"=== word clusters ===\")\nword_clusters = [val for key, val in doc.spans.items() if key.startswith(\"coref_head\")]\nfor cluster in word_clusters:\n    print(cluster)\n# check the expanded clusters\nprint(\"=== full clusters ===\")\nfull_clusters = [val for key, val in doc.spans.items() if key.startswith(\"coref_cluster\")]\nfor cluster in full_clusters:\n    print(cluster)\n```\n\nThe prefixes used here are a user setting, so you can customize them for your\nown pipelines.\n"}
```
Comment (rmitsch) [inline projects.jsonl:16]: Did you lint/validate the `.jsonl` file to make sure it's valid? It looks alright to me at first glance, I'm just confused as to why the GH syntax highlighting doesn't work in some lines.

Comment (polm) [reply projects.jsonl:16]: Confirmed it's OK with jq. I GH just doesn't highlight very long lines, for example this is a little over 1000 "a"s:

```jsonl
{"ok": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"}
```

**Comment by rmitsch** [comment]
From https://github.com/explosion/projects/pull/160:
> Another reason is that currently the script just fails with an error on some projects, which needs to be checked separately.

Has this been fixed?

**Comment by polm** [comment]
That is a reference to `update_configs.py`, not this script. This script has no errors that I'm aware of.

File: projects.jsonl (hunk 4 above)
Comment (rmitsch) [reply projects.jsonl:16]: Thanks for checking!

**Review by rmitsch — APPROVED** [review]

File: .github/update_projects_jsonl.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        path = path.parent
+
+        # prep data for the json file
+        config = load_project_config(path)
+        entry = {"shortname": f"{path.parent.name}/{path.name}"}
+        entry["title"] = config["title"]
+        entry["description"] = config.get("description", "")
+        entries.append(entry)
+
+    with open("projects.jsonl", "w") as jsonfile:
```
Comment (adrianeboyd) [inline .github/update_projects_jsonl.py:28]:
```suggestion
    with open("projects.jsonl", "w", encoding="utf-8") as jsonfile:
```

**Pull request merged**